File: Readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)

{
    string text = get_string("Text: ");  // get user input
    printf("Grade ");
    int j = 0;
    int q = 0;
    int w = 0;
    for (int c = 0, n = strlen(text); c < n; c++) // get string length
    {
        if (isalpha(text[c])) // get letter count
        {
            j++;
        }
        if (text[c] == '.' || text[c] == '!' || text[c] == '?') // get sentence count
        {
            q++;
        }
        if (isblank(text[c])) // get word count
        {
            w++;
        }
    }
    float correctw = w + 1;

    float avgj = (j / correctw) * 100; // calc avg letters per 100 words

    float avgq = (q / correctw) * 100; // calc avg sentences per 100 words

    int index = round(0.0588 * avgj - 0.296 * avgq - 15.8); // calc and round grade level

    if (index < 1) // print results
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("%i\n", index);
    }
}

File: Readability.py
from cs50 import get_string

text = get_string("Text: ")
j = 0.0
q = 0.0
w = 1.0
for i in range(len(text)):
    if ((text[i] >= 'a' and text[i] <= 'z') or (text[i] >= 'A' and text[i] <= 'Z')):
        j += 1.0
    if text[i] == '.' or text[i] == '!' or text[i] == '?':
        q += 1.0
    if text[i] == ' ':
        w += 1.0

j = ((j / w) * 100.0)
q = ((q / w) * 100.0)
index = round(0.0588 * j - 0.296 * q - 15.8)

if (index < 1):
    print("Before Grade 1")
elif (index > 16):
    print("Grade 16+")
else:
    print(f"Grade: {index}")
